### PR TITLE
5060 upgrade virtualenv

### PR DESCRIPTION
**Pull request by kroustou**

### CODE REVIEW

File: packagingbuild/Dockerfile.template [hunk 1]
```diff
@@ -122,7 +122,7 @@ RUN echo -e "#!/bin/sh\nexit 101\n" > /usr/sbin/policy-rc.d && \
 {% if (version == '8') -%}
 # Install development tools and python3 for EL8
  RUN yum -y install python3 python3-devel rpmdevtools python3-virtualenv && \
-     pip3 install virtualenv==16.6.0 pip==19.1.1 wheel setuptools
+     pip3 install virtualenv==20.2.2 pip==20.3.3 wheel setuptools
```
Comment (amanda11) [inline packagingbuild/Dockerfile.template:125]: Need to also amend the line further down that still pins virtualenv to 16.6.0 for el7

**Review by amanda11 — CHANGES_REQUESTED** [review]

**Comment by amanda11** [comment]
I'm also seeing problems if I use the image built from these changes to run st2-packages and build. It then fails when it runs:
/usr/bin/python3 ../scripts/fixate-requirements.py -s in-requirements.txt -f ../fixed-requirements.txt

which looks to be because of the change:
```
[bdist: st2api]      /usr/bin/python3 ../scripts/fixate-requirements.py -s in-requirements.txt -f ../fixed-requirements.txt
[bdist: st2api]      Traceback (most recent call last):
[bdist: st2api]        File "../scripts/fixate-requirements.py", line 213, in <module>
[bdist: st2api]          skip=skip)
[bdist: st2api]        File "../scripts/fixate-requirements.py", line 144, in write_requirements
[bdist: st2api]          requirements = merge_source_requirements(sources)
[bdist: st2api]        File "../scripts/fixate-requirements.py", line 121, in merge_source_requirements
[bdist: st2api]          if req.req:
[bdist: st2api]      AttributeError: 'ParsedRequirement' object has no attribute 'req'
[bdist: st2api]      make: *** [.stamp-requirements] Error 1
```

File: packagingbuild/Dockerfile.template (hunk 1 above)
Comment (kroustou) [reply packagingbuild/Dockerfile.template:125]: Done, thanks, did not catch that one.

**Comment by kroustou** [comment]
Okay waiting on your feedback. Thank you and sorry for the delays i have been really busy lately :)

**Comment by armab** [comment]
As @amanda11 identified, the pip/virtualenv upgrades might be more involved, your work is proceeding here https://github.com/StackStorm/st2/pull/5123 and https://github.com/StackStorm/st2packaging-dockerfiles/pull/98.

Thanks @kroustou for starting this story!

**Comment by kroustou** [comment]
Thank you very much for your hard work

**Comment by amanda11** [comment]
@kroustou I think this has been superseded by changes committed to master under PR https://github.com/StackStorm/st2packaging-dockerfiles/pull/98

**Pull request closed without merging**